File: 2315-count-asterisks/2315-count-asterisks.go
func countAsterisks(s string) int {
	pair := false
	var count int;
	for i := 0; i < len(s); i++ {
		switch s[i] {
		case '|':
			pair = !pair
		case '*':
			if !pair {
				count++
			}
		}
	}
	return count
}

File: 2315-count-asterisks/2315-count-asterisks.java
class Solution {
    public int countAsterisks(String s) {
        int count = 0;
        boolean pair = false;
        for (int i = 0; i < s.length(); i++) {
            switch (s.charAt(i)) {
                case '|':
                    pair = !pair;
                    break;
                case '*':
                    if (!pair) {
                        count++;
                    }
            }
        }
        return count;
    }
}